leif/leifd/leifapi/filters/filters.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filter
+
+import (
+	drghs_v1 "github.com/GoogleCloudPlatform/devrel-services/drghs/v1"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
+)
+
+const defaultFilter = "true"
+
+// Owner checks if the Owner passes the given CEL expression.
+func Owner(o *drghs_v1.Owner, filter string) (bool, error) {
+	if filter == "" {
+		filter = defaultFilter
+	}
+	if o == nil {
+		return false, nil
+	}
+
+	env, err := cel.NewEnv(
+		cel.Types(&drghs_v1.Owner{}),
+		cel.Declarations(
+			decls.NewIdent("owner", decls.NewObjectType("drghs.v1.Owner"), nil)))
+
+	parsed, issues := env.Parse(filter)
+	if issues != nil && issues.Err() != nil {
+		return false, issues.Err()
+	}
+	checked, issues := env.Check(parsed)
+	if issues != nil && issues.Err() != nil {
+		return false, issues.Err()
+	}
+	prg, err := env.Program(checked)
+	if err != nil {
+		return false, err
+	}
```

Comment:
The process of building a CEL Program is... shockingly expensive. Since this will be called O(n) times and the program will not change this will be more performant if it is broken out into two functions, an `BuildOwnerFilter` Function which return `(cel.Program, error)` and a `FilterOwner(o Owner, p Program)` function. 

`BuildOwnerFilter` is called only once per filter string and then FilterOwner is called N times while iterating over the set of Owners.